helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -375,13 +375,15 @@ public void removeDeploymentGroup(final String name) throws DeploymentGroupDoesN
     final ZooKeeperClient client = provider.get("removeDeploymentGroup");
     try {
       client.ensurePath(Paths.configDeploymentGroups());
-      client.delete(Paths.configDeploymentGroup(name));
+
       if (client.exists(Paths.statusDeploymentGroupHosts(name)) != null) {
         client.delete(Paths.statusDeploymentGroupHosts(name));
       }
       if (client.exists(Paths.statusDeploymentGroup(name)) != null) {
         client.delete(Paths.statusDeploymentGroup(name));
       }
+
+      client.delete(Paths.configDeploymentGroup(name));
```

Comment:
if the issue was that this path didn't exist (not clear to me if that was the case), why not wrap it in an `if (client.exists(path))` block like the other deletes?